### PR TITLE
Tries to address the refresh problem of the table in the dialog "Manage external file types" (issue 5902)

### PR DESCRIPTION
**Pull request by systemoperator**
References: #5902

Tries to address the visual issue, where the table in the dialog "Manage external file types" (Options > Manage external file types) is only refreshed, when an updated entry firstly gets scrolled out of the visible table area and then gets scrolled back in, so that the table entry is again visible, then also the values contained in this changed entry is actually shown updated.

Currently, it does not fix it, but maybe it is useful and/or should be merged anyway, since other JavaFX tables are implemented in the same way, by using the extractor property. You decide.

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)

### CODE REVIEW

File: src/main/java/org/jabref/gui/externalfiletype/UnknownExternalFileType.java [hunk 1]
```diff
@@ -11,36 +14,46 @@
...
 
     @Override
-    public String getName() {
+    public StringProperty getName() {
         return name;
     }
 
     @Override
-    public String getExtension() {
+    public String getNameAsString() {
```
Comment (koppor) [inline src/main/java/org/jabref/gui/externalfiletype/UnknownExternalFileType.java]: Some code uses `.getValue()` some `...AsString()` I would always stick to `.getValue()`

File: src/main/java/org/jabref/gui/externalfiletype/CustomizeExternalFileTypesViewModel.java [hunk 2]
```diff
@@ -30,11 +33,11 @@ public void storeSettings() {
     public void resetToDefaults() {
         List<ExternalFileType> list = ExternalFileTypes.getDefaultExternalFileTypes();
         fileTypes.setAll(list);
-        fileTypes.sort(Comparator.comparing(ExternalFileType::getName));
+        fileTypes.sort(Comparator.comparing(ExternalFileType::getNameAsString));
     }
 
     public void addNewType() {
-        CustomExternalFileType type = new CustomExternalFileType("", "", "", "", "new", IconTheme.JabRefIcons.FILE);
+        CustomExternalFileType type = new CustomExternalFileType(new SimpleStringProperty(""), new SimpleStringProperty(""), new SimpleStringProperty(""), new SimpleStringProperty(""), new SimpleStringProperty("new"), IconTheme.JabRefIcons.FILE);
```
Comment (koppor) [inline src/main/java/org/jabref/gui/externalfiletype/CustomizeExternalFileTypesViewModel.java]: Can't a string be passed and converted in `CustomExternalFileTypes` internally?

File: src/main/java/org/jabref/gui/externalfiletype/ExternalFileType.java [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
 import org.jabref.model.entry.field.FieldFactory;
 
 public interface ExternalFileType {
-    String getName();
+    StringProperty getName();
+
+    String getNameAsString();
 
-    String getExtension();
+    StringProperty getExtension();
```
Comment (koppor) [inline src/main/java/org/jabref/gui/externalfiletype/ExternalFileType.java]: Where is the `StringProperty` used? I always saw `getValue()`. Did I miss something?

**Comment by Siedlerchr** [comment]
I don't this is the right approach with passing the StringProperty down, I would rather create a ViewModel for the table like I did in the Custom entry types dialog and not modifying the original implementation of the model type.

**Comment by systemoperator** [comment]
I would recommend, that @Siedlerchr fixes this issue as needed. He has (positive) experience concerning this matter. :)

**Comment by koppor** [comment]
@systemoperator This is your chance to learn about MVVM. See https://devdocs.jabref.org/javafx#architecture-model-view-controller-viewmodel-mv-c-vm for details. I hope, you will resume work here.

**Comment by tobiasdiez** [comment]
I agree with https://github.com/JabRef/jabref/pull/5907#issuecomment-588153566. @systemoperator it would be nice if you could implement this! Thanks.

**Comment by Siedlerchr** [comment]
I also looked into the original code, and I could also reproduce this under Windows and I guess it's due to the Singleton stuff updating.

**Comment by koppor** [comment]
I see that MVVM is hard to implement. @Siedlerchr will take care later. We will close for now as we have to focus on the 5.1 release 🔥

**Pull request closed without merging**